File: codewars/6kyu/sort-the-odd/dduck-sort-the-odd.js
// sort-the-odd
// https://www.codewars.com/kata/578aa45ee9fd15ff4600090d

const sortArray = (array) => {
  const oddNumbers = array.filter((v) => v % 2 === 1).sort((a, b) => a - b);
  return array.map((v) => v % 2 === 1 ? oddNumbers.shift() : v);
};


console.log(sortArray([5, 3, 2, 8, 1, 4]), [1, 3, 2, 8, 5, 4]);
console.log(sortArray([5, 3, 1, 8, 0]), [1, 3, 5, 8, 0]);
console.log(sortArray([]), []);

File: codewars/6kyu/sort-the-odd/dduck-sort-the-odd.py
# sort-the-odd
# https://www.codewars.com/kata/578aa45ee9fd15ff4600090d

from unittest import TestCase

from collections import deque


def sort_array(source_array):
    odd_num = deque(sorted(filter(lambda x: x % 2 == 1, source_array)))
    return list(map(lambda x: odd_num.popleft() if x % 2 == 1 else x, source_array))


TestCase().assertEqual(sort_array([5, 3, 2, 8, 1, 4]), [1, 3, 2, 8, 5, 4])
TestCase().assertEqual(sort_array([5, 3, 1, 8, 0]), [1, 3, 5, 8, 0])
TestCase().assertEqual(sort_array([]), [])
